src/CXXTests/testSidreStorage.cc

```diff
@@ -0,0 +1,475 @@
+#include "Field/Field.hh"
+#include "Utilities/SidreDataCollection.hh"
+#include "Geometry/Dimension.hh"
+#include "axom/sidre.hpp"
+
+#include "gtest/gtest.h"
+#include <iostream>
+
+template <typename T>
+class SidreDataCollectionTest : public ::testing::Test
+{
+  public:
+    Spheral::SidreDataCollection myData;
+    int n = 100;
+    T* rawSidreData;
+
+    Spheral::Field<Spheral::Dim<1>, T> makeField()
+    {
+      Spheral::NodeList<Spheral::Dim<1>> makeNodeList("test bed", n, 0);
+      Spheral::Field<Spheral::Dim<1>, T> testField("test field", makeNodeList);
+      for (int i = 0; i < n; i++)
+        testField[i] = i;
+      return testField;
+    }
+
+    void allocRawSidreData(const Spheral::Field<Spheral::Dim<1>, T>& testField)
+    {
+      rawSidreData = myData.alloc_view("SidreTest", testField)->getData();
+    }
+};
```

Comment:
Let's move all helper classes, functions, includes and type aliases into a separate header file keeping only the gtest "TEST" type calls in this file.